A29-xds-tls-security.md

```diff
@@ -0,0 +1,509 @@
+[server-authz]: #server-authorization-aka-subject-alt-name-checks
+[VCCP1]: https://github.com/envoyproxy/envoy/blob/7d4b2cae486b66b62ba0d3e1e348504699bea1bf/api/envoy/extensions/transport_sockets/tls/v3/tls.proto#L193
+[VCSSC1]: https://github.com/envoyproxy/envoy/blob/7d4b2cae486b66b62ba0d3e1e348504699bea1bf/api/envoy/extensions/transport_sockets/tls/v3/tls.proto#L184
+
+### Use of SPIFFE IDs in Certificates
+
+To facilitate service mesh secure communication, we support [SPIFFE][] based certificates and peer certificate
+verification following the [SPIFFE][] specification and the [match_subject_alt_names][] field in xDS.
+
+[SPIFFE]: https://github.com/spiffe/spiffe
+[match_subject_alt_names]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-field-extensions-transport-sockets-tls-v3-certificatevalidationcontext-match-subject-alt-names
+
+### Certificate Provider Plugin Framework
+
+xDS supports many different configurations for endpoints to obtain certificates but gRPC supports only
+`CertificateProviderInstance`. Through this object the control plane tells the xDS client the name of
+a certificate and the "instance" name of a provider to use to obtain a certificate. This provider
+instance name is translated into a provider implementation and a configuration for that implementation
+using the client's own configuration instead of being sent by the control plane. This enables flexibility
+in heterogeneous deployments where different clients can use different implementations for the same
+provider instance name. For example, an on-prem client may get its certificate using a local certificate
+provider whereas a client running in cloud may use a cloud-vendor provided implementation to obtain
+the certificate.
+
+gRPC provides a `CertificateProvider` plugin API that can support multiple implementations. The plugin API
+is not currently public, so applications cannot currently add their own provider implementations although
+we might make this API public in the future. However gRPC currently includes a `file_watcher` provider
+implementation - descibed [below][FILE-WATCHER-LINK] - which reads certificates from the local file system.
+
+[FILE-WATCHER-LINK]: #file_watcher-certificate-provider
+
+Certificate provider plugin instances are configured via the xDS bootstrap file.  There is a new
+top-level field called `"certificate_providers"` whose value is a map (a JSON object).  The key
+in this map is the certificate_provider_instance name and the value is a JSON object
+having exactly 2 fields:
+
+* `"plugin_name"` which is the name of the plugin implementation (a string value), and
+* `"config"` which is the configuration for the plugin. The value of `"config"` is a JSON object
+whose schema is defined by that plugin.
+
+The structure in the bootstrap file is as follows:
+
+```
```

Comment:
```suggestion
```jsonc
```

hint: use `jsonc` to render json with comments:

```jsonc
"certificate_providers": {
  "instance_name": {
    "plugin_name": "implementation_name",
    "config": {
      // ...config for implementation_name plugin...
    }
  }
}
```